src/Microsoft.OpenApi/Models/OpenApiMediaType.cs

```diff
@@ -43,6 +43,23 @@ public class OpenApiMediaType : IOpenApiSerializable, IOpenApiExtensible
         /// </summary>
         public IDictionary<string, IOpenApiExtension> Extensions { get; set; } = new Dictionary<string, IOpenApiExtension>();
 
+        /// <summary>
+        /// Parameterless constructor
+        /// </summary>
+        public OpenApiMediaType() {}
+
+        /// <summary>
+        /// Initializes a copy of an <see cref="OpenApiMediaType"/> object
+        /// </summary>
+        public OpenApiMediaType(OpenApiMediaType mediaType)
+        {
+            Schema = new(mediaType.Schema);
+            Example = mediaType.Example;
```

Comment:
new ? but to what type? maybe get the constructor by reflection?

Reply:
Maybe here we need to make IOpenApiAny support ICloneable to support this particular case.

Reply:
adding a new interface to the IOpenApiAny definition will be a breaking change for implementers outside of this library. Maggie started down the path of trying to find a copy constructor by reflection and defaulting back to a reference copy which I believe will be more flexible for implementers.